support-models/src/main/scala/com/gu/salesforce/Salesforce.scala

```diff
@@ -202,7 +202,8 @@ object Salesforce {
   }
 
   object Authentication {
-    implicit val codec: Codec[Authentication] = deriveCodec
+    implicit val decodeDateTime: Decoder[DateTime] = Decoder.decodeLong.map(new DateTime(_))
+    implicit val codec: Decoder[Authentication] = deriveDecoder
```

Comment:
```
{
    "access_token": "asjhdgksajhdgaj",
    "instance_url": "https://aksjhdg--DEV.my.salesforce.com",
    "id": "https://test.salesforce.com/id/kjasghdkjashdjkasd/askjdhakjsdhkajshdka",
    "token_type": "Bearer",
    "issued_at": "1603976679073",
    "signature": "dskjfhlskjdfhlksdjfhlksdjfhlksdjhf"
}
```

Reply:
You can import a `Decoder[DateTime]` from `CustomCodecs`

Reply:
Could we leave the most commonly used one (probably the long version?) in there and move the other version out to where it's needed? I think it's good to have at least one version in there because it is the obvious place to look for it in the codebase

Reply:
I'm not sure I could work out a most commonly used one, it's always determined based on what you need for that purpose and we use each of the 4 once!  I'd prefer where possible to use ISO most commonly as it's human readable.
- the long encoder is only used once to serialise a promo date to the client side.
- the long decoder is used to decode the salesforce credential expiry time
- the ISO decoder is used to get promo start/end dates from dynamo
- the ISO encoder is used to push the delivery date to membership-workflow (and in a dynamo test)

Actually, I think based on your comments I will actually put both in there, but each in a suitably named nested object.  That means people have to choose explicitly but it's easy to find.   There was somewhere I realised was using the wrong one so it makes sense to be explicit.

The other option is to not serialise DateTime directly, but convert it to a String or Long in the case class.  But I think that will make the case classes less type safe and add boiler plate at the point of use.

Reply:
I like this approach 👍